app/src/main/java/de/tum/in/tumcampusapp/activities/wizard/WizNavCheckTokenActivity.java

```diff
@@ -87,22 +88,26 @@ protected Integer onLoadInBackground(Void... arg) {
             }
         } else {
             // Get users full name
-            TUMOnlineRequest<IdentitySet> request2 = new TUMOnlineRequest<>(TUMOnlineConst.Companion.getIDENTITY(), this, true);
-            Optional<IdentitySet> id = request2.fetch();
+            TUMOnlineRequest<IdentitySet> request = new TUMOnlineRequest<>(TUMOnlineConst.Companion.getIDENTITY(), this, true);
+            Optional<IdentitySet> id = request.fetch();
             if (!id.isPresent()) {
                 return R.string.no_rights_to_access_id;
             }
 
+            Identity identity = id.get()
+                                  .getIds()
+                                  .get(0);
+
             // Save the name to preferences
-            Utils.setSetting(this, Const.CHAT_ROOM_DISPLAY_NAME, id.get()
-                                                                   .toString());
+            Utils.setSetting(this, Const.CHAT_ROOM_DISPLAY_NAME, identity
+                    .toString());
 
             // Save the TUMOnline id to preferences
-            String pID = getUserPIdentNr(id.get()
-                                           .toString());
-            if (pID != null) {
-                Utils.setSetting(this, Const.TUMO_PIDENT_NR, pID);
-            }
+            Utils.setSetting(this, Const.TUMO_PIDENT_NR, identity.getObfuscatd_id());
+            Utils.setSetting(this, Const.TUMO_ST_NR, identity.getObfuscatd_ids().getStudierende());
```

Comment:
Could we please not use those cryptic TUMonline identifiers, but readable names?